src/conda_package_handling/conda_fmt.py

```diff
@@ -47,47 +32,69 @@ def supported(fn):
     @staticmethod
     def extract(fn, dest_dir, **kw):
         components = utils.ensure_list(kw.get("components")) or ("info", "pkg")
-        file_id = os.path.basename(fn).replace(".conda", "")
         if not os.path.isabs(fn):
             fn = os.path.normpath(os.path.join(os.getcwd(), fn))
         if not os.path.isdir(dest_dir):
             os.makedirs(dest_dir)
-        for component in components:
-            _extract_component(fn, file_id, component, dest_dir)
+
+        _extract(str(fn), str(dest_dir), components=components)
 
     @staticmethod
     def extract_info(fn, dest_dir=None):
         return CondaFormat_v2.extract(fn, dest_dir, components=["info"])
 
     @staticmethod
-    def create(prefix, file_list, out_fn, out_folder=os.getcwd(), **kw):
+    def create(
+        prefix,
+        file_list,
+        out_fn,
+        out_folder=os.getcwd(),
+        compressor=lambda: zstandard.ZstdCompressor(
+            level=ZSTD_COMPRESS_LEVEL, threads=ZSTD_COMPRESS_THREADS
+        ),
+        **kw,
+    ):
         if os.path.isabs(out_fn):
             out_folder = os.path.dirname(out_fn)
             out_fn = os.path.basename(out_fn)
         conda_pkg_fn = os.path.join(out_folder, out_fn)
-        out_fn = out_fn.replace(".conda", "")
+        file_id = out_fn = out_fn.replace(".conda", "")
         pkg_files = utils.filter_info_files(file_list, prefix)
         info_files = set(file_list) - set(pkg_files)
-        ext, comp_filter, filter_opts = kw.get("compression_tuple") or DEFAULT_COMPRESSION_TUPLE
 
-        with utils.TemporaryDirectory(dir=out_folder) as tmpdir:
-            info_tarball = create_compressed_tarball(
-                prefix, info_files, tmpdir, "info-" + out_fn, ext, comp_filter, filter_opts
-            )
-            pkg_tarball = create_compressed_tarball(
-                prefix, pkg_files, tmpdir, "pkg-" + out_fn, ext, comp_filter, filter_opts
+        # legacy libarchive-ish compatibility
+        ext, comp_filter, filter_opts = compression_tuple = kw.get(
+            "compression_tuple", (None, None, None)
+        )
+        if filter_opts and filter_opts.startswith("zstd:compression-level="):
+            compressor = lambda: zstandard.ZstdCompressor(
+                level=int(filter_opts.split("=", 1)[-1]),
+                threads=ZSTD_COMPRESS_THREADS,
             )
 
+        with ZipFile(conda_pkg_fn, "w", compression=ZIP_STORED) as conda_file, utils.tmp_chdir(
+            prefix
+        ):
+
             pkg_metadata = {"conda_pkg_format_version": CONDA_PACKAGE_FORMAT_VERSION}
+            conda_file.writestr("metadata.json", json.dumps(pkg_metadata))
+
+            # put the info last, for parity with updated transmute.
+            for component, files in (f"pkg-{file_id}.tar.zst", pkg_files), (
+                f"info-{file_id}.tar.zst",
+                info_files,
+            ):
```

Comment:
```suggestion
            components_and_files = (
                (f"pkg-{file_id}.tar.zst", pkg_files),
                (f"info-{file_id}.tar.zst", info_files)
            ) 
            for component, files in components_and_files:
```

Please resist the temptation to put everything on a single line. It really makes things hard to read.